src/main/java/uk/gov/justice/digital/job/DataHubBatchJob.java

```diff
@@ -144,14 +145,10 @@ private void processFilePaths(SparkSession sparkSession, String schema, String t
                 logger.warn("No source reference for table {}.{} - writing all data to violations", schema, table);
                 violationService.handleNoSchemaFoundS3(sparkSession, dataFrame, schema, table, STRUCTURED_LOAD);
             }
-        } catch (SparkException e) {
```

Comment:
I've verified that SparkException is what is actually thrown both via stacktrace and debugging while running the app against the dev environment from intellij. But it just isn't caught when running the app using the jar. I'm fairly certain that the reason is Java optimising away code paths it thinks can never be reached.